kernel: bump 5.4 to 5.4.115

graysky2 opened the pull request:
All patches automatically rebased.
```
Build system: x86_64
Build-tested: ipq806x/R7800
Run-tested: ipq806x/R7800
```
No dmesg regressions, everything functional

Signed-off-by: John Audia <graysky@archlinux.us>

graysky2 commented:
Thanks for the help rebasing `700-v5.5-net-core-allow-fast-GRO-for-skbs-with-Ethernet-heade.patch` @solbjorn

hauke commented on target/linux/generic/backport-5.4/700-v5.5-net-core-allow-fast-GRO-for-skbs-with-Ethernet-heade.patch:
```diff
@@ -66,13 +66,43 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
+ 
+ 	skb_reset_mac_header(skb);
+-	skb_gro_reset_offset(skb);
++	skb_gro_reset_offset(skb, hlen);
```
Why is this hlen needed here?
Kernel 5.10.32 contains also both patches, but just ```skb_gro_reset_offset(skb)``` is called here.

graysky2 replied:
@solbjorn - Can you address this question?

solbjorn replied:
@hauke, @graysky2, we decided to also include this upstream fix: https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net.git/commit/?id=7ad18ff6449cbd6beb26b53128ddf56d2685aa93
It will land mainline soon, the reason why it's important is because Eric Dumazet's check for NET_IP_ALIGN causes huge slowdowns on drivers which use `napi_gro_frags()`.

hauke replied:
Thank you for the link.
I would prefer if you add this as an extra patch into the backports-5.4 folder and not squash multiple upstream commits together here.

graysky2 replied:
@hauke - I am happy to do so (@solbjorn actually provided them to me as two discrete patches).  I am but am ignorant as to how to add a patch.  Do I simply place the new file in `target/linux/generic/backport-5.4`?

graysky2 replied:
@hauke - Done, please see the new commit force pushed.

timocapa commented:
```
Build system: x86_64
Build-tested: ramips/mt7621/{Xiaomi R3G, Redmi 2100}, Lantiq/XWAY/Fritz!Box 7320/7330
Run-tested: ramips/mt7621/{Xiaomi R3G, Redmi 2100}, Lantiq/XWAY/Fritz!Box 7320/7330
```
Looks good, sorry I've been absent for a while.

graysky2 committed:
kernel: bump 5.4 to 5.4.114

Manually rebased*
  generic/backport-5.4/700-v5.5-net-core-allow-fast-GRO-for-skbs-with-Ethernet-heade.patch

Added new backport*
  generic/backport-5.4/050-gro-fix-napi_gro_frags-Fast-GRO-breakage-due-to-IP-a.patch

All others updated automatically.

The new backport was included based on this[1] upstream commit that will be
mainlined soon.  This change is needed because Eric Dumazet's check for
NET_IP_ALIGN (landed in 5.4.114) causes huge slowdowns on drivers which use
napi_gro_frags().

Build system: x86_64
Build-tested: ipq806x/R7800
Run-tested: ipq806x/R7800

No dmesg regressions, everything functional

*Credit to Alexander Lobakin
1. https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net.git/commit/?id=7ad18ff6449cbd6beb26b53128ddf56d2685aa93

Signed-off-by: John Audia <graysky@archlinux.us>

graysky2 committed:
kernel: bump 5.4 to 5.4.115

All patches automatically rebased.

Build system: x86_64
Build-tested: ipq806x/R7800
Run-tested: ipq806x/R7800

No dmesg regressions, everything functional

Signed-off-by: John Audia <graysky@archlinux.us>

graysky2 commented:
@timocapa - Can you verify 5.4.115 is good on your hardware?

timocapa commented:
> @timocapa - Can you verify 5.4.115 is good on your hardware?

Sent this right after I went to sleep (EU/Germany) lol

Sure, give me a bit

timocapa commented:
```
Build system: x86_64
Build-tested: ramips/mt7621/{Xiaomi R3G, Redmi 2100}
Run-tested: ramips/mt7621/{Xiaomi R3G, Redmi 2100}
```

The Fritzbox decided to lose connection to my network so i couldnt update it, and I'm too lazy to reboot it right now, but both ramips seem fine.

hauke commented:
Thank you for the pull request. I applied it to master.
